src/csharp/Microsoft.Spark/Sql/Types/Timestamp.cs

```diff
@@ -0,0 +1,140 @@
+    /// Coordinated Universal Time (UTC).
+    /// </summary>
+    public class Timestamp
+    {
+        private readonly DateTime _dateTime;
+
+        /// <summary>
+        /// Constructor for Timestamp class.
+        /// </summary>
+        /// <param name="dateTime">DateTime object</param>
+        public Timestamp(DateTime dateTime)
+        {
+            _dateTime = dateTime.ToUniversalTime();
+        }
+
+        /// <summary>
+        /// Constructor for Timestamp class, the defaut timezone is
+        /// Coordinated Universal Time (UTC).
+        /// </summary>
+        /// <param name="year">The year (1 through 9999)</param>
+        /// <param name="month">The month (1 through 12)</param>
+        /// <param name="day">The day (1 through the number of days in month)</param>
+        /// <param name="hour">The hour (0 through 23)</param>
+        /// <param name="minute">The minute (0 through 59)</param>
+        /// <param name="second">The second (0 through 59)</param>
+        /// <param name="microsecond">The microsecond (0 through 999999)</param>
+        public Timestamp(
+            int year,
+            int month,
+            int day,
+            int hour,
+            int minute,
+            int second,
+            int microsecond)
+        {
+            if (microsecond <= 999999 && microsecond >= 0)
+            {
+                // Create DateTime and AddTicks based on the microsecond value.
+                _dateTime = new DateTime(year, month, day, hour, minute, second, DateTimeKind.Utc)
+                    .AddTicks(microsecond * 10);
+            }
+            else
+            {
+                throw new ArgumentOutOfRangeException($"Invalid microsecond value {microsecond}. " +
+                    $"The microsecond is from 0 through 999999.");
+            }
+        }
+
+        /// <summary>
+        /// Returns the year component of the timestamp.
+        /// </summary>
+        public int Year => _dateTime.Year;
+
+        /// <summary>
+        /// Returns the month component of the timestamp.
+        /// </summary>
+        public int Month => _dateTime.Month;
+
+        /// <summary>
+        /// Returns the day component of the timestamp.
+        /// </summary>
+        public int Day => _dateTime.Day;
+
+        /// <summary>
+        /// Returns the hour component of the timestamp.
+        /// </summary>
+        public int Hour => _dateTime.Hour;
+
+        /// <summary>
+        /// Returns the minute component of the timestamp.
+        /// </summary>
+        public int Minute => _dateTime.Minute;
+
+        /// <summary>
+        /// Returns the second component of the timestamp.
+        /// </summary>
+        public int Second => _dateTime.Second;
+
+        /// <summary>
+        /// Returns the microsecond component of the timestamp.
+        /// </summary>
+        public int Microsecond => (int)(_dateTime.Ticks -
```

Comment:
Can't you just do `(int)(_dateTime.Ticks % 10000000 / 10)`?

Reply:
If we use `(int)(_dateTime.Ticks % 10000000 / 10)`, it will be the whole value represented in microseconds instead of just microsecond value, no?

Reply:
Since we are modding, it should just get the microsecond value.

Reply:
Yep, right! Thanks.